src/interface/src/app/auth.service.ts

```diff
@@ -2,24 +2,32 @@ import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { CanActivate, Router } from '@angular/router';
 import { CookieService } from 'ngx-cookie-service';
-import { catchError, map, Observable, of, shareReplay, Subject, tap } from 'rxjs';
+import { BehaviorSubject, catchError, map, Observable, of, shareReplay, Subject, tap } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthService {
 
-  loggedInStatus$: Subject<boolean> = new Subject();
-  isLoggedIn$: Observable<boolean> = this.loggedInStatus$.pipe(shareReplay(1));
+  loggedInStatus$: Subject<boolean> = new BehaviorSubject(false);
+  isLoggedIn$: Observable<boolean> = this.loggedInStatus$;
 
   private readonly API_ROOT = 'http://localhost:8000/dj-rest-auth/';
 
   constructor(
     private http: HttpClient,
-    private cookieService: CookieService,
-    private router: Router) {
+    private cookieService: CookieService) {
       this.isLoggedIn$.subscribe();
```

Comment:
I'm not an expert on rxjs, but I read in documentation that "An Observable instance begins publishing values only when someone subscribes to it." Without the initial subscription here, the other components watching the Observable didn't get the published values.

Reply:
I *think* it should be subscribed in the component that's using it rather than the service (src/interface/src/app/account-dialog/account-dialog.component.ts)
There's not a good way to close subscriptions in the service if it's info you're emitting out to the components. It'll keep persisting the observable (it won't know when the component doesn't need it anymore).
https://dev.to/ussdlover/best-practice-for-subscribing-to-observables-in-services-with-angular-1712